scripts/static-urlify-readme-images.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Our READMEs contain images. But we don't want to publish images in NPM
+// tarballs, because we want to keep them small. This script replaces any
+// relative-path image references in the READMEs with static
+// githubusercontent.com URLs, using the SHA of master.
+
+import {execFileSync} from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import {URL} from 'url';
+
+const mwcRepoRoot = path.resolve(__dirname, '..', '..');
+
+const githubRaw =
+    `https://raw.githubusercontent.com/material-components/material-components-web-components`;
+
+// Matches markdown image syntax like `![description](url "tooltip")`
+const imageRegexp = /(\!\[.*?\]\()([^) ]+)(.*?\))/g;
```

Comment:
why is `?` required in the last capture group?

Reply:
It makes it non-greedy, otherwise it would match `![description](url "tooltip")any subsequent characters as long as there's eventually another )`

Reply:
Could also be `([^)]*\))`, but either is fine with me